### PR TITLE
core/validatorapi: workaround for Gnosis proposals

### PR DESCRIPTION
**Pull request by gsora**
If the "gnosis_block_hotfix" feature flag is enabled, use the Gnosis-compatible representation of the proposal when calculating its hash tree root.  This should fix Gnosis proposals failing due to bad signature.

category: feature
ticket: #3181 
feature_flag: gnosis_block_hotfix

### CODE REVIEW

File: core/signeddata.go [hunk 1]
```diff
@@ -320,6 +322,12 @@ func (p VersionedSignedProposal) MessageRoot() ([32]byte, error) {
 			return p.DenebBlinded.Message.HashTreeRoot()
 		}
 
+		if p.IsGnosis {
```
Comment (pinebit) [inline core/signeddata.go:325]: Wondering if start dealing with blinded blocks, shouldn't this `IsGnosis` come first, before `.Blinded` check?

Comment (pinebit) [reply core/signeddata.go:325]: What if instead of this flag we simply test for `featureset.Enabled(featureset.GnosisBlockHotfix)`?

**Review by pinebit — APPROVED** [review]
Approved with comments.
Plus consider adding that "real block" unit test.

File: core/signeddata.go (hunk 1 above)
Comment (gsora) [reply core/signeddata.go]: Implemented!

File: go.mod [hunk 2]
```diff
@@ -205,3 +205,7 @@ require (
 
 // We're replacing kryptology with our own fork, which fixes dependencies' security vulnerabilities.
 replace github.com/coinbase/kryptology => github.com/ObolNetwork/kryptology v0.0.0-20231016091344-eed023b6cac8
+
+// We're replacing go-eth2-client with a branch off our fork, at version v0.21.10.
+// This is needed to ensure Gnosis compatibility.
+replace github.com/attestantio/go-eth2-client => github.com/ObolNetwork/go-eth2-client v0.21.11-0.20240822135044-f0a5b21e02c6
```
Comment (KaloyanTanev) [inline go.mod:211]: Wouldn't that mean we will have to manage this repo constantly and rebase it every time there is a new version of go-eth2-client?

Comment (gsora) [reply go.mod:211]: Hopefully go-eth2-client will fix this upstream soon

**Review by KaloyanTanev — APPROVED** [review]

**Pull request merged**